mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+
+            else:
+                print("Error, dimension not read correctly")
+
+        plt.show()
+
+
+class make_animation:
+    """
+    make_plot plots contents of McStasData objects
+
+    Plotting is controlled through options assosciated with the
+    McStasData objects.  If a list is given, the plots appear in one
+    subplot.
+    """
+
+    def __init__(self, data_list, **kwargs):
+        """
+        plots McStasData, single object or list of McStasData
+
+        The options concerning plotting are stored with the data
+
+        Parameters
+        ----------
+        data_list : McStasData or list of McStasData
+            McStasData to be plotted
+        """
+        if not isinstance(data_list, McStasData):
+            print("number of elements in data list = "
+                  + str(len(data_list)))
+        else:
+            # Make list from single element to simplify syntax
+            data_list = [data_list]
+
+        # Relevant options:
+        #  select colormap
+        #  show / hide colorbar
+        #  custom title / label
+        #  color of 1d plot
+        #  overlay several 1d
+        #  log scale (o$rders of magnitude)
+        #  compare several 1d
+        #  compare 2D
+
+        if "fontsize" in kwargs:
+            plt.rcParams.update({'font.size': kwargs["fontsize"]})
+            
+        if "fps" in kwargs:
+            period_in_ms = 1000/kwargs["fps"]
+        else:
+            period_in_ms = 200
+
+        fig = plt.figure()
+        ax = plt.axes()
+        #fig, ax = plt.subplot()
+        
+        # find limits for entire dataset
+        maximum_values = []
+        minimum_values = []
+        
+        is_1D = False
+        is_2D = False
+        
+        for data in data_list:
+            if isinstance(data.metadata.dimension, int):
+                is_1D = True
+                
+                y = data.Intensity[np.nonzero(data.Intensity)]
+                if len(y) > 0:
+                    maximum_values.append(y.max())
+                    minimum_values.append(y.min())
+            
+            elif len(data.metadata.dimension) == 2:
+                is_2D = True
+                
+                y = data.Intensity[np.nonzero(data.Intensity)]
+                if len(y) > 0:
+                    maximum_values.append(y.max())
+                    minimum_values.append(y.min())
+        
+        if len(maximum_values) > 0:
+            maximum_value = np.array(maximum_values).max()
+        else:
+            maximum_value = 0
+            
+        if len(minimum_values) > 0:
+            minimum_value = np.array(minimum_values).min()
+        else:
+            minimum_value = 0
+        
+        if is_1D and is_2D:
+            raise InputError(
+                "Both 1D and 2D data in animation, only one allowed.")
+
+        # initialize plots
+        
+        data = data_list[0]
+        if isinstance(data.metadata.dimension, int):
```

Comment:
Again i would try to just call `make_plot` here instead of duplicating again.

Reply:
Made a internal plotting function instead that make_plot, make_sub_plot and make_animation all use. The amount of duplication was really bad, glad to be rid of it!